Makefile

```diff
@@ -14,11 +14,32 @@
 # limitations under the License.
 # -----------------------------------------------------------------------
 
+# do not indent (Makefile syntax, not bash)
+ifdef MVN_RELEASE_VERSION
+# maven release
+VERSION=$(MVN_RELEASE_VERSION)
+else
+# not a maven release
+ifdef POM_VERSION
+VERSION=$(POM_VERSION)
+endif
+endif
+# one of these will be set to VERSION in Jenkins Production Environment
+
+
+ifndef VERSION
+# local build, use the POM version
+VERSION:=$(shell mvn -q -Dexec.executable=echo -Dexec.args='$${project.version}' --non-recursive exec:exec)
```

Comment:
Why do we have two `$` characters?

Reply:
Because the expression inside parentheses needs to get evaluated and assigned to variable `VERSION`.